### PR TITLE
Fix #672 from XSK issues

### LINKED ISSUE #672: [Templates] MongoDB based Persistence for Application Templates
Implement a MongoDB based Persistence template for Application generation

### PR DESCRIPTION
**Pull request by dimitarquanterall**
### What does this PR do?

Adds import / export functionality anywhere within the working tree. 

### What issues does this PR fix or reference?

Fixes: https://github.com/SAP/xsk/issues/672

#### Release Notes

Works in combination with https://github.com/eclipse/dirigible/pull/1664

### CODE REVIEW

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 1]
```diff
@@ -1318,6 +1347,79 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+            // console.info('onErrorItem', fileItem, response, status, headers);
+        };
+        uploader.onCancelItem = function (fileItem, response, status, headers) {
+            // console.info('onCancelItem', fileItem, response, status, headers);
+        };
+        uploader.onCompleteItem = function (fileItem, response, status, headers) {
+            // console.info('onCompleteItem', fileItem, response, status, headers);
+        };
+        uploader.onCompleteAll = function () {
+            // console.log('COMPLETE UPLOAD');
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1408]: Please remove the comment.

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 2]
```diff
@@ -1318,6 +1347,79 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+
+        uploader.onWhenAddingFileFailed = function (item /*{File|FileLikeObject}*/, filter, options) {
+            // console.info('onWhenAddingFileFailed', item, filter, options);
+        };
+        uploader.onAfterAddingFile = function (fileItem) {
+
+        };
+        uploader.onAfterAddingAll = function (addedFileItems) {
+            // console.info('onAfterAddingAll', addedFileItems);
+        };
```
Comment (ThuF) [inline ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js:1380]: Remove the unused, not implemented functions, if not needed.

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 3]
```diff
@@ -1318,6 +1347,79 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+            // console.log('COMPLETE UPLOAD');
+            this.clearQueue();
+            $('#uploadZipToFolder').click();
+            $('#refreshButton').click();
+        };
+
+        this.cancelImportFromZip = function (uploader) {
+            if (uploader.isUploading)
+                uploader.cancelAll();
+            uploader.clearQueue();
```
[the remark this answers is not in the document]
Comment (dimitarquanterall) [reply ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js]: Fixed in a new commit

File: ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js [hunk 4]
```diff
@@ -1318,6 +1347,79 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
...
+        };
+        uploader.onErrorItem = function (fileItem, response, status, headers) {
+            // console.info('onErrorItem', fileItem, response, status, headers);
+        };
+        uploader.onCancelItem = function (fileItem, response, status, headers) {
+            // console.info('onCancelItem', fileItem, response, status, headers);
+        };
+        uploader.onCompleteItem = function (fileItem, response, status, headers) {
+            // console.info('onCompleteItem', fileItem, response, status, headers);
+        };
```
[the remark this answers is not in the document]
Comment (dimitarquanterall) [reply ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js]: They are not needed now. Removed in a new commit

**Review by ThuF — APPROVED** [review]

**Pull request merged**